README.md

```diff
@@ -84,6 +84,45 @@ jobs:
           channelId: live
 ```
 
+### Authenticate using Workload Identity Federation
+
+Managing service account keys poses a security risk. Workload Identity Federation can be used to reduce the risk when running GitHub Actions.  
+With identity federation, GitHub will generate a JWT that will be used to authenticate against Google Cloud APIs.
+
+Before setting up your GitHub Action, you need to follow these steps to prepare your Workload Identity Pool:  
+https://github.com/google-github-actions/auth?tab=readme-ov-file#workload-identity-federation-through-a-service-account
+
+```yaml
+jobs:
+  deploy_using_workload_identity_federatoin:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write' # This permission is needed
+    steps:
+      - uses: actions/checkout@v2
+
+      # Add these two steps to generate the credential to use with the `action-hosting-deploy` action.
+      - name: Prepare Google Cloud authentication 
```

Comment:
This step seems to be missing `id: auth` otherwise I think `${{ steps.auth.outputs.credentials_file_path }}` won't work below.

Reply:
```suggestion
      - name: Prepare Google Cloud authentication 
        id: auth
```